nav2_controller/src/controller_server.cpp

```diff
@@ -719,16 +730,18 @@ void ControllerServer::publishVelocity(const geometry_msgs::msg::TwistStamped &
 
 void ControllerServer::publishZeroVelocity()
 {
-  geometry_msgs::msg::TwistStamped velocity;
-  velocity.twist.angular.x = 0;
-  velocity.twist.angular.y = 0;
-  velocity.twist.angular.z = 0;
-  velocity.twist.linear.x = 0;
-  velocity.twist.linear.y = 0;
-  velocity.twist.linear.z = 0;
-  velocity.header.frame_id = costmap_ros_->getBaseFrameID();
-  velocity.header.stamp = now();
-  publishVelocity(velocity);
+  if (get_parameter("publish_zero_velocity").as_bool()) {
```

Comment:
You can store the variable in the class, that's OK

Reply:
According to https://www.ros.org/reps/rep-0009.html:

```
You cannot...
  Add new non-static data members, even if they are protected or private
```

Also the `get_parameter` is a map lookup so should be fast.

Reply:
I suppose I haven't enforced that part of ABI, but perhaps I should be -- it would just make direct backporting _any new feature_ virtually impossible since they would likely have some parameter set tied to them. I enforce API stability and behavior stability (i.e. not changing defaults or fundamentally changing the behavior on a module mid-distribution), but I think there's an argument that I should by enforcing stricter ABI as well. Though, its never come up and no one's had an issue. 

This is fine.